Update active_storage_overview.md [ci skip]

trebeil opened the pull request:
Adds information about how to pass additional arguments to transformations.

<!--
Thanks for contributing to Rails!

Please do not make *Draft* pull requests, as they still send
notifications to everyone watching the Rails repo.

Create a pull request when it is ready for review and feedback
from the Rails team :).

If your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

About this template

The following template aims to help contributors write a good description for their pull requests.
We'd like you to provide a description of the changes in your pull request (i.e. bugs fixed or features added), the motivation behind the changes, and complete the checklist below before opening a pull request.

Feel free to discard it if you need to (e.g. when you just fix a typo). -->

### Motivation / Background

This Pull Request has been created because the right syntax for passing optional arguments for transformation methods in an attachment variant call is not documented anywhere and it is not intuitive. Yet, it is probably needed by many people who use ActiveStorage to create variants of attachments.

### Detail

This Pull Request adds a few lines to the ActiveStorage guide showing how to pass additional arguments to a transformation method in an attachment variant call.

### Additional information

Doesn't apply.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

trebeil committed:
Update active_storage_overview.md [ci-skip]

Adds information about how to pass additional arguments to transformations.

Co-authored-by: Vipul A M <vipul@hey.com>

vipulnsward commented on guides/source/active_storage_overview.md:
````diff
@@ -961,6 +961,11 @@ location.
 <%= image_tag user.avatar.variant(resize_to_limit: [100, 100]) %>
 ```
 
+Some transformations accept optional arguments. Pass any additional arguments to the transformation function as key: value pairs inside a hash:
````
```suggestion
Some transformations accept optional arguments. Pass any additional arguments to the transformation function as key: value pairs inside a hash:

```

skipkayhil committed:
Reflow Active Storage transforms section [ci-skip]

Both "optional arguments" and "migrating processors" seem too specific
for where they were previously located within the section. Both ideas
are more complicated/processor specific, but are located very early on
when the ideas should be more simple/introductory.

This commit reorders the paragraphs a bit to improve the section's flow.
Now, the most basic examples are presented first, then the idea of
processors and what they have in common, and finally the more advanced
options and how the processors differ.

skipkayhil commented:
Thanks for the PR!
